internal/pkg/cli/deploy/workload.go

```diff
@@ -393,18 +416,47 @@ func (img ContainerImageIdentifier) Tag() string {
 }
 
 func (d *workloadDeployer) uploadContainerImages(out *UploadArtifactsOutput) error {
-	// If it is built from local Dockerfile, build and push to the ECR repo.
-	buildArgsPerContainer, err := buildArgsPerContainer(d.name, d.workspacePath, d.image, d.mft)
+	return buildAndPushImages(&ImageActionInput{
+		Name:               d.name,
+		WorkspacePath:      d.workspacePath,
+		Image:              d.image,
+		Mft:                d.mft,
+		CustomTag:          d.image.CustomTag,
+		GitShortCommitTag:  d.image.GitShortCommitTag,
+		Builder:            d.repository,
+		Login:              d.repository.Login,
+		CheckDockerEngine:  d.docker.CheckDockerEngineRunning,
+		LabeledTermPrinter: d.labeledTermPrinter,
+	}, out)
+
+}
+
+// BuildContainerImages builds the all the images given the build arguments
+func BuildContainerImages(in *ImageActionInput, out *UploadArtifactsOutput) error {
+	return buildImages(in, out)
+}
+
+func buildImages(in *ImageActionInput, out *UploadArtifactsOutput) error {
+	return performImageAction(in, out, in.Builder.Build)
+}
+
+func buildAndPushImages(in *ImageActionInput, out *UploadArtifactsOutput) error {
+	return performImageAction(in, out, in.Builder.BuildAndPush)
+}
```

Comment:
seems like we could get rid of these two functions and just call `performImageAction()` directly from `BuildContainerImages()` and `uploadContainerImages()`